### PR TITLE
Make transactions receipts part of transaction inclusion proof

### PR DESCRIPTION
**Pull request by svyatonik**
(we only support EIP-658 transactions)

### CODE REVIEW

File: relays/ethereum/src/ethereum_exchange.rs [hunk 1]
```diff
@@ -221,11 +227,18 @@ impl TargetClient<EthereumToSubstrateExchange> for SubstrateTransactionsTarget {
 
 	async fn filter_transaction_proof(&self, proof: &EthereumTransactionInclusionProof) -> Result<bool, Self::Error> {
 		// let's try to parse transaction locally
-		let parse_result = bridge_node_runtime::exchange::EthTransaction::parse(&proof.proof[proof.index as usize]);
+		let (raw_tx, raw_tx_receipt) = &proof.proof[proof.index as usize];
```
Comment (tomusdrw) [inline relays/ethereum/src/ethereum_exchange.rs:230]: Do we verify somewhere that `proof.index < proof.proof.len()`?

File: primitives/ethereum-poa/src/lib.rs [hunk 2]
```diff
@@ -325,6 +333,25 @@ impl UnsignedTransaction {
 }
 
 impl Receipt {
+	/// Decode status from raw transaction receipt RLP.
+	pub fn is_successful_raw_receipt(raw_receipt: &[u8]) -> Result<bool, DecoderError> {
+		let rlp = Rlp::new(raw_receipt);
+		if rlp.item_count()? == 3 {
+			// no outcome - invalid tx?
+			Ok(false)
```
[the remark this answers is not in the document]
Comment (svyatonik) [reply primitives/ethereum-poa/src/lib.rs:341]: I've took this from [OE](https://github.com/openethereum/openethereum/blob/master/ethcore/types/src/receipt.rs#L91) - in this case outcome is `Unknown`, which we treat as 'failed'. TBH - I'm not sure what `Unknown` really means, but for us (EIP-658-only) it is 'failed' iiuc.

File: relays/ethereum/src/ethereum_exchange.rs (hunk 1 above)
Comment (svyatonik) [reply relays/ethereum/src/ethereum_exchange.rs:230]: We verify it where it comes from untrusted source - i.e. when verifying proof in runtime module. Here it is relay talking to relay, so it is kind of trusted.

File: primitives/ethereum-poa/src/lib.rs [hunk 3]
```diff
@@ -325,6 +333,25 @@ impl UnsignedTransaction {
...
+		} else {
+			let first = rlp.at(0)?;
+			if first.is_data() && first.data()?.len() <= 1 {
```
[the remark this answers is not in the document]
Comment (svyatonik) [reply primitives/ethereum-poa/src/lib.rs:344]: It is from OE code as well - I could add test for this case, if you think it worth it.

[the remark this answers is not in the document]
Comment (tomusdrw) [reply primitives/ethereum-poa/src/lib.rs:344]: Yes, let's do that please. We can alternatively just match `== 1` instead. I just don't understand why we want `.len() == 0` to fall into this branch.

[the remark this answers is not in the document]
Comment (svyatonik) [reply primitives/ethereum-poa/src/lib.rs:344]: I've added test `is_successful_raw_receipt_with_empty_data` - `as_val()` isn't failing, but returns 0. Which means that transaction has failed && we'll return `Ok(false)`.

**Pull request merged**